src/crystal/syntax_highlighter/html.cr

```diff
@@ -0,0 +1,98 @@
+require "../syntax_highlighter"
+
+# A syntax highlighter that renders Crystal source code with HTML markup.
+#
+# ```
+# code = %(foo = bar("baz\#{PI + 1}") # comment)
+# html = Crystal::SyntaxHighlighter::HTML.highlight(code)
+# html # => "foo <span class=\"o\">=</span> bar(<span class=\"s\">&quot;baz</span><span class=\"i\">\#{</span><span class=\"t\">PI</span> <span class=\"o\">+</span> <span class=\"n\">1</span><span class=\"i\">}</span><span class=\"s\">&quot;</span>) <span class=\"c\"># comment</span>"
+# ```
+class Crystal::SyntaxHighlighter::HTML < Crystal::SyntaxHighlighter
+  # Highlights *code* and writes the result to *io*.
+  def self.highlight(io : IO, code : String)
+    new(io).highlight(code)
+  end
+
+  # Highlights *code* and returns the result.
+  def self.highlight(code : String)
+    String.build do |io|
+      highlight(io, code)
+    end
+  end
+
+  # Highlights *code* or returns unhighlighted *code* on error.
+  #
+  # Same as `.highlight(code : String)` except that any error is rescued and
+  # returns unhighlighted source code.
+  def self.highlight!(code : String)
+    highlight(code)
+  rescue
+    code
```

Comment:
I believe it is invalid to return the original string because there's an expectation that the output will be escaped HTML. So the output needs to be escaped here. + a spec for this.

Reply:
And seems [`ca3ef39` (#11366)](https://github.com/crystal-lang/crystal/pull/11366/commits/ca3ef39140e79c36e0c0259649572d8a00025b42) should be reverted them